core/src/test/java/io/kestra/core/runners/pebble/filters/ChunkFilterTest.java

```diff
@@ -38,4 +38,32 @@ void exception() {
             }).get();
         });
     }
+    @Test
+    void chunkWithIntegerVariable() throws IllegalVariableEvaluationException {
+        // Reproducer for issue: Integer variable causing ClassCastException
+        Map<String, Object> vars = Map.of(
+            "max_items", 2,  // This is an Integer, not a Long
```

Comment:
```suggestion
            "max_items", Integer.valueOf(2), 
```

Lilke that we are sure that it's and Integer